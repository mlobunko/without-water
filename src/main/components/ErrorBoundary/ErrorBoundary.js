import React from 'react';

export class ErrorBoundary extends React.Component {
  state = { hasError: false };

  componentDidCatch = () => {
    this.setState({ hasError: true });
  };

  render() {
    if (this.state.hasError) {
      return <h1>Something went wrong.</h1>;
    }
    return this.props.children;
  }
}
ErrorBoundary.displayName = 'ErrorBoundary';

export default ErrorBoundary;
