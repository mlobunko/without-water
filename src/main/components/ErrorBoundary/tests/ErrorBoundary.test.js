import React from 'react';
import { shallow } from 'enzyme';

import { ErrorBoundary } from '../ErrorBoundary';

describe('ErrorBoundary component', () => {
  it('componentDidCatch works correctly', () => {
    const wrapper = shallow(<ErrorBoundary>someText</ErrorBoundary>);
    wrapper.setState({ hasError: false });
    wrapper.instance().componentDidCatch();
    expect(wrapper.state().hasError).toEqual(true);
  });

  it('renders error message when there is an error', () => {
    const wrapper = shallow(<ErrorBoundary>someText</ErrorBoundary>);
    wrapper.setState({ hasError: true });
    expect(wrapper.text()).toEqual('Something went wrong.');
  });

  it('renders properly when there is an error', () => {
    const wrapper = shallow(<ErrorBoundary>someText</ErrorBoundary>);
    wrapper.setState({ hasError: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders text when there is no error', () => {
    const wrapper = shallow(<ErrorBoundary>someText</ErrorBoundary>);
    wrapper.setState({ hasError: false });
    expect(wrapper.text()).toEqual('someText');
  });

  it('renders properly when there is no error', () => {
    const wrapper = shallow(<ErrorBoundary>someText</ErrorBoundary>);
    wrapper.setState({ hasError: false });
    expect(wrapper).toMatchSnapshot();
  });
});
