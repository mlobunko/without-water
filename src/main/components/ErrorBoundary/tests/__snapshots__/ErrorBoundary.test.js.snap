// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component renders properly when there is an error 1`] = `
<h1>
  Something went wrong.
</h1>
`;

exports[`ErrorBoundary component renders properly when there is no error 1`] = `"someText"`;
